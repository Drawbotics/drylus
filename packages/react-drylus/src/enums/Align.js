import Enum from '@drawbotics/enums';


const Align = new Enum(
  'LEFT',
  'CENTER',
  'RIGHT',
);


export default Align;
