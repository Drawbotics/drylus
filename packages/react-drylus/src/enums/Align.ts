export enum Align {
  LEFT = 'LEFT',
  CENTER = 'CENTER',
  RIGHT = 'RIGHT',
  JUSTIFY = 'JUSTIFY',
}
