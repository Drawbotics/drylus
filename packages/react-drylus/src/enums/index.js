export Category from './Category';
export Tier from './Tier';
export Size from './Size';
export Position from './Position';
export Align from './Align';
export Color from './Color';
export Shade from './Shade';
