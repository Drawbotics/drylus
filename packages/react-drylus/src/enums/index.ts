export * from './Category';
export * from './Tier';
export * from './Size';
export * from './Position';
export * from './Align';
export * from './Color';
export * from './Shade';
