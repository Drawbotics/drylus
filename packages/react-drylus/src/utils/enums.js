import camelCase from 'lodash/camelCase';

import { Category, Color, Shade, Tier } from '../enums';

export function getEnumAsClass(enumVal) {
  return camelCase(enumVal?.description?.toLowerCase());
}

export function colorEnumToCategory(enumVal) {
  switch (enumVal) {
    case Color.RED:
      return Category.DANGER;
    case Color.GREEN:
      return Category.SUCCESS;
    case Color.ORANGE:
      return Category.WARNING;
    case Color.BLUE:
      return Category.INFO;
  }
  return enumVal;
}

export function categoryEnumToColor(enumVal) {
  switch (enumVal) {
    case Category.DANGER:
      return Color.RED;
    case Category.SUCCESS:
      return Color.GREEN;
    case Category.WARNING:
      return Color.ORANGE;
    case Category.INFO:
      return Color.BLUE;
  }
  return enumVal;
}

export function shadeEnumToTier(enumVal) {
  switch (enumVal) {
    case Shade.DARK:
      return Tier.PRIMARY;
    case Shade.MEDIUM:
      return Tier.SECONDARY;
    case Shade.LIGHT:
      return Tier.TERTIARY;
  }
  return enumVal;
}

export function getIconForCategory(category) {
  switch (category) {
    case Category.DANGER:
      return 'alert-circle';
    case Category.SUCCESS:
      return 'check-circle';
    case Category.WARNING:
      return 'alert-triangle';
    default:
      return 'info';
  }
}
