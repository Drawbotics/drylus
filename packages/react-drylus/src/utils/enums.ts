import camelCase from 'lodash/camelCase';

import { IconType } from '../components';
import { Category, Color, Position, Shade, Size, Tier } from '../enums';

type Enum = Category | Color | Shade | Tier | Size | Position;

export function getEnumAsClass<T extends Record<string, string>>(enumVal?: Enum): keyof T {
  return camelCase(enumVal?.toLowerCase());
}

export function colorEnumToCategory(enumVal: Color): Category | undefined {
  switch (enumVal) {
    case Color.RED:
      return Category.DANGER;
    case Color.GREEN:
      return Category.SUCCESS;
    case Color.ORANGE:
      return Category.WARNING;
    case Color.BLUE:
      return Category.INFO;
    case Color.BRAND:
      return Category.BRAND;
    case Color.PRIMARY:
      return Category.PRIMARY;
  }
}

export function categoryEnumToColor(enumVal: Category): Color | undefined {
  switch (enumVal) {
    case Category.DANGER:
      return Color.RED;
    case Category.SUCCESS:
      return Color.GREEN;
    case Category.WARNING:
      return Color.ORANGE;
    case Category.INFO:
      return Color.BLUE;
    case Category.BRAND:
      return Color.BRAND;
    case Category.PRIMARY:
      return Color.PRIMARY;
  }
}

export function shadeEnumToTier(enumVal: Shade): Tier | undefined {
  switch (enumVal) {
    case Shade.DARK:
      return Tier.PRIMARY;
    case Shade.MEDIUM:
      return Tier.SECONDARY;
    case Shade.LIGHT:
      return Tier.TERTIARY;
  }
}

export function getIconForCategory(category: Category): IconType {
  switch (category) {
    case Category.DANGER:
      return 'alert-circle';
    case Category.SUCCESS:
      return 'check-circle';
    case Category.WARNING:
      return 'alert-triangle';
    default:
      return 'info';
  }
}
