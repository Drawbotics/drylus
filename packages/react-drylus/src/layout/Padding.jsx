import React from 'react';
import PropTypes from 'prop-types';
import { css, cx } from 'emotion';
import sv from '@drawbotics/drylus-style-vars';
import camelCase from 'lodash/camelCase';

import Sizes from '../base/Sizes';
import { useResponsiveProps } from '../utils/hooks';


const styles = {
  root: css`
    padding: ${sv.defaultPadding};

    @media ${sv.screenL} {
      padding: ${sv.paddingSmall};
    }
  `,
  resetPadding: css`
    padding: 0;

    @media ${sv.screenL} {
      padding: 0;
    }
  `,
  extraSmall: css`
    padding: ${sv.paddingExtraSmall};

    @media ${sv.screenL} {
      padding: calc(${sv.paddingExtraSmall} / 2);
    }
  `,
  small: css`
    padding: ${sv.paddingSmall};

    @media ${sv.screenL} {
      padding: ${sv.paddingExtraSmall};
    }
  `,
  large: css`
    padding: ${sv.paddingLarge};

    @media ${sv.screenL} {
      padding: ${sv.defaultPadding};
    }
  `,
  extraLarge: css`
    padding: ${sv.paddingExtraLarge};

    @media ${sv.screenL} {
      padding: ${sv.paddingLarge};
    }
  `,
  extraSmallLeft: css`
    padding-left: ${sv.paddingExtraSmall};

    @media ${sv.screenL} {
      padding-left: calc(${sv.paddingExtraSmall} / 2);
    }
  `,
  extraSmallRight: css`
    padding-right: ${sv.paddingExtraSmall};

    @media ${sv.screenL} {
      padding-right: calc(${sv.paddingExtraSmall} / 2);
    }
  `,
  extraSmallTop: css`
    padding-top: ${sv.paddingExtraSmall};

    @media ${sv.screenL} {
      padding-top: calc(${sv.paddingExtraSmall} / 2);
    }
  `,
  extraSmallBottom: css`
    padding-bottom: ${sv.paddingExtraSmall};

    @media ${sv.screenL} {
      padding-bottom: calc(${sv.paddingExtraSmall} / 2);
    }
  `,
  smallLeft: css`
    padding-left: ${sv.paddingSmall};

    @media ${sv.screenL} {
      padding-left: ${sv.paddingExtraSmall};
    }
  `,
  smallRight: css`
    padding-right: ${sv.paddingSmall};

    @media ${sv.screenL} {
      padding-right: ${sv.paddingExtraSmall};
    }
  `,
  smallTop: css`
    padding-top: ${sv.paddingSmall};

    @media ${sv.screenL} {
      padding-top: ${sv.paddingExtraSmall};
    }
  `,
  smallBottom: css`
    padding-bottom: ${sv.paddingSmall};

    @media ${sv.screenL} {
      padding-bottom: ${sv.paddingExtraSmall};
    }
  `,
  defaultLeft: css`
    padding-left: ${sv.defaultPadding};

    @media ${sv.screenL} {
      padding-left: ${sv.paddingSmall};
    }
  `,
  defaultRight: css`
    padding-right: ${sv.defaultPadding};

    @media ${sv.screenL} {
      padding-right: ${sv.paddingSmall};
    }
  `,
  defaultTop: css`
    padding-top: ${sv.defaultPadding};

    @media ${sv.screenL} {
      padding-top: ${sv.paddingSmall};
    }
  `,
  defaultBottom: css`
    padding-bottom: ${sv.defaultPadding};

    @media ${sv.screenL} {
      padding-bottom: ${sv.paddingSmall};
    }
  `,
  largeLeft: css`
    padding-left: ${sv.paddingLarge};

    @media ${sv.screenL} {
      padding-left: ${sv.defaultPadding};
    }
  `,
  largeRight: css`
    padding-right: ${sv.paddingLarge};

    @media ${sv.screenL} {
      padding-right: ${sv.defaultPadding};
    }
  `,
  largeTop: css`
    padding-top: ${sv.paddingLarge};

    @media ${sv.screenL} {
      padding-top: ${sv.defaultPadding};
    }
  `,
  largeBottom: css`
    padding-bottom: ${sv.paddingLarge};

    @media ${sv.screenL} {
      padding-bottom: ${sv.defaultPadding};
    }
  `,
  extraLargeLeft: css`
    padding-left: ${sv.paddingExtraLarge};

    @media ${sv.screenL} {
      padding-left: ${sv.paddingLarge};
    }
  `,
  extraLargeRight: css`
    padding-right: ${sv.paddingExtraLarge};

    @media ${sv.screenL} {
      padding-right: ${sv.paddingLarge};
    }
  `,
  extraLargeTop: css`
    padding-top: ${sv.paddingExtraLarge};

    @media ${sv.screenL} {
      padding-top: ${sv.paddingLarge};
    }
  `,
  extraLargeBottom: css`
    padding-bottom: ${sv.paddingExtraLarge};

    @media ${sv.screenL} {
      padding-bottom: ${sv.paddingLarge};
    }
  `,
};


const Padding = ({
  responsive,
  ...rest,
}) => {
  const { children, size, style } = useResponsiveProps(rest, responsive);

  const isUniform = typeof size !== 'object';
  return (
    <div className={cx(styles.root, {
      [styles[camelCase(size?.description)]]: isUniform && size,
      [styles.resetPadding]: ! isUniform,
      [styles[camelCase(`${size?.left?.description}_LEFT`)]]: ! isUniform && size?.left,
      [styles[camelCase(`${size?.right?.description}_RIGHT`)]]: ! isUniform && size?.right,
      [styles[camelCase(`${size?.top?.description}_TOP`)]]: ! isUniform && size?.top,
      [styles[camelCase(`${size?.bottom?.description}_BOTTOM`)]]: ! isUniform && size?.bottom,
    })} style={style}>
      {children}
    </div>
  );
};


Padding.propTypes = {
  /** Determines the amount of padding given to the component. If a single value, the padding is applied equally to each side */
  size: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.oneOf([ Sizes.DEFAULT, Sizes.SMALL, Sizes.EXTRA_SMALL, Sizes.LARGE, Sizes.EXTRA_LARGE ]),
    PropTypes.shape({
      left: PropTypes.oneOf([ Sizes.DEFAULT, Sizes.SMALL, Sizes.EXTRA_SMALL, Sizes.LARGE, Sizes.EXTRA_LARGE ]),
      right: PropTypes.oneOf([ Sizes.DEFAULT, Sizes.SMALL, Sizes.EXTRA_SMALL, Sizes.LARGE, Sizes.EXTRA_LARGE ]),
      bottom: PropTypes.oneOf([ Sizes.DEFAULT, Sizes.SMALL, Sizes.EXTRA_SMALL, Sizes.LARGE, Sizes.EXTRA_LARGE ]),
      top: PropTypes.oneOf([ Sizes.DEFAULT, Sizes.SMALL, Sizes.EXTRA_SMALL, Sizes.LARGE, Sizes.EXTRA_LARGE ]),
    }),
  ]),

  /** The content of the padding wrapper */
  children: PropTypes.node,

  /** Prop to override any style if necessary, use sparingly */
  style: PropTypes.object,

  /** Reponsive prop overrides */
  responsive: PropTypes.shape({
    XS: PropTypes.object,
    S: PropTypes.object,
    M: PropTypes.object,
    L: PropTypes.object,
    XL: PropTypes.object,
    HUGE: PropTypes.object,
  }),
};


export default Padding;
