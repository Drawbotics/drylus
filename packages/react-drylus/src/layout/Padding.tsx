import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import camelCase from 'lodash/camelCase';
import React from 'react';

import { Size } from '../enums';
import { Rectangular, Responsive, Style, Variable } from '../types';
import { useResponsiveProps } from '../utils';

const styles = {
  root: css`
    padding: ${sv.defaultPadding};

    @media ${sv.screenL} {
      padding: ${sv.paddingSmall};
    }
  `,
  resetPadding: css`
    padding: 0;

    @media ${sv.screenL} {
      padding: 0;
    }
  `,
  extraSmall: css`
    padding: ${sv.paddingExtraSmall};

    @media ${sv.screenL} {
      padding: calc(${sv.paddingExtraSmall} / 2);
    }
  `,
  small: css`
    padding: ${sv.paddingSmall};

    @media ${sv.screenL} {
      padding: ${sv.paddingExtraSmall};
    }
  `,
  large: css`
    padding: ${sv.paddingLarge};

    @media ${sv.screenL} {
      padding: ${sv.defaultPadding};
    }
  `,
  extraLarge: css`
    padding: ${sv.paddingExtraLarge};

    @media ${sv.screenL} {
      padding: ${sv.paddingLarge};
    }
  `,
  huge: css`
    padding: ${sv.paddingHuge};

    @media ${sv.screenL} {
      padding: ${sv.paddingExtraLarge};
    }
  `,
  extraHuge: css`
    padding: ${sv.paddingExtraHuge};

    @media ${sv.screenL} {
      padding: ${sv.paddingHuge};
    }
  `,
  massive: css`
    padding: ${sv.paddingMassive};

    @media ${sv.screenL} {
      padding: ${sv.paddingExtraHuge};
    }
  `,
  extraSmallLeft: css`
    padding-left: ${sv.paddingExtraSmall};

    @media ${sv.screenL} {
      padding-left: calc(${sv.paddingExtraSmall} / 2);
    }
  `,
  extraSmallRight: css`
    padding-right: ${sv.paddingExtraSmall};

    @media ${sv.screenL} {
      padding-right: calc(${sv.paddingExtraSmall} / 2);
    }
  `,
  extraSmallTop: css`
    padding-top: ${sv.paddingExtraSmall};

    @media ${sv.screenL} {
      padding-top: calc(${sv.paddingExtraSmall} / 2);
    }
  `,
  extraSmallBottom: css`
    padding-bottom: ${sv.paddingExtraSmall};

    @media ${sv.screenL} {
      padding-bottom: calc(${sv.paddingExtraSmall} / 2);
    }
  `,
  smallLeft: css`
    padding-left: ${sv.paddingSmall};

    @media ${sv.screenL} {
      padding-left: ${sv.paddingExtraSmall};
    }
  `,
  smallRight: css`
    padding-right: ${sv.paddingSmall};

    @media ${sv.screenL} {
      padding-right: ${sv.paddingExtraSmall};
    }
  `,
  smallTop: css`
    padding-top: ${sv.paddingSmall};

    @media ${sv.screenL} {
      padding-top: ${sv.paddingExtraSmall};
    }
  `,
  smallBottom: css`
    padding-bottom: ${sv.paddingSmall};

    @media ${sv.screenL} {
      padding-bottom: ${sv.paddingExtraSmall};
    }
  `,
  defaultLeft: css`
    padding-left: ${sv.defaultPadding};

    @media ${sv.screenL} {
      padding-left: ${sv.paddingSmall};
    }
  `,
  defaultRight: css`
    padding-right: ${sv.defaultPadding};

    @media ${sv.screenL} {
      padding-right: ${sv.paddingSmall};
    }
  `,
  defaultTop: css`
    padding-top: ${sv.defaultPadding};

    @media ${sv.screenL} {
      padding-top: ${sv.paddingSmall};
    }
  `,
  defaultBottom: css`
    padding-bottom: ${sv.defaultPadding};

    @media ${sv.screenL} {
      padding-bottom: ${sv.paddingSmall};
    }
  `,
  largeLeft: css`
    padding-left: ${sv.paddingLarge};

    @media ${sv.screenL} {
      padding-left: ${sv.defaultPadding};
    }
  `,
  largeRight: css`
    padding-right: ${sv.paddingLarge};

    @media ${sv.screenL} {
      padding-right: ${sv.defaultPadding};
    }
  `,
  largeTop: css`
    padding-top: ${sv.paddingLarge};

    @media ${sv.screenL} {
      padding-top: ${sv.defaultPadding};
    }
  `,
  largeBottom: css`
    padding-bottom: ${sv.paddingLarge};

    @media ${sv.screenL} {
      padding-bottom: ${sv.defaultPadding};
    }
  `,
  extraLargeLeft: css`
    padding-left: ${sv.paddingExtraLarge};

    @media ${sv.screenL} {
      padding-left: ${sv.paddingLarge};
    }
  `,
  extraLargeRight: css`
    padding-right: ${sv.paddingExtraLarge};

    @media ${sv.screenL} {
      padding-right: ${sv.paddingLarge};
    }
  `,
  extraLargeTop: css`
    padding-top: ${sv.paddingExtraLarge};

    @media ${sv.screenL} {
      padding-top: ${sv.paddingLarge};
    }
  `,
  extraLargeBottom: css`
    padding-bottom: ${sv.paddingExtraLarge};

    @media ${sv.screenL} {
      padding-bottom: ${sv.paddingLarge};
    }
  `,
  hugeLeft: css`
    padding-left: ${sv.paddingHuge};

    @media ${sv.screenL} {
      padding-left: ${sv.paddingExtraLarge};
    }
  `,
  hugeRight: css`
    padding-right: ${sv.paddingHuge};

    @media ${sv.screenL} {
      padding-right: ${sv.paddingExtraLarge};
    }
  `,
  hugeTop: css`
    padding-top: ${sv.paddingHuge};

    @media ${sv.screenL} {
      padding-top: ${sv.paddingExtraLarge};
    }
  `,
  hugeBottom: css`
    padding-bottom: ${sv.paddingHuge};

    @media ${sv.screenL} {
      padding-bottom: ${sv.paddingExtraLarge};
    }
  `,
  extraHugeLeft: css`
    padding-left: ${sv.paddingExtraHuge};

    @media ${sv.screenL} {
      padding-left: ${sv.paddingHuge};
    }
  `,
  extraHugeRight: css`
    padding-right: ${sv.paddingExtraHuge};

    @media ${sv.screenL} {
      padding-right: ${sv.paddingHuge};
    }
  `,
  extraHugeTop: css`
    padding-top: ${sv.paddingExtraHuge};

    @media ${sv.screenL} {
      padding-top: ${sv.paddingHuge};
    }
  `,
  extraHugeBottom: css`
    padding-bottom: ${sv.paddingExtraHuge};

    @media ${sv.screenL} {
      padding-bottom: ${sv.paddingHuge};
    }
  `,
  massiveLeft: css`
    padding-left: ${sv.paddingMassive};

    @media ${sv.screenL} {
      padding-left: ${sv.paddingExtraHuge};
    }
  `,
  massiveRight: css`
    padding-right: ${sv.paddingMassive};

    @media ${sv.screenL} {
      padding-right: ${sv.paddingExtraHuge};
    }
  `,
  massiveTop: css`
    padding-top: ${sv.paddingMassive};

    @media ${sv.screenL} {
      padding-top: ${sv.paddingExtraHuge};
    }
  `,
  massiveBottom: css`
    padding-bottom: ${sv.paddingMassive};

    @media ${sv.screenL} {
      padding-bottom: ${sv.paddingExtraHuge};
    }
  `,
};

function _computeSize(sizeDescription: Rectangular) {
  let size: Variable = {};

  if (sizeDescription.horizontal) {
    size.left = sizeDescription.horizontal;
    size.right = sizeDescription.horizontal;
  }
  if (sizeDescription.vertical) {
    size.top = sizeDescription.vertical;
    size.bottom = sizeDescription.vertical;
  }

  return size;
}

function _isRectangular(size: any): size is Rectangular {
  return size.vertical || size.horizontal;
}

export interface PaddingProps {
  /** Determines the amount of padding given to the component. If a single value, the padding is applied equally to each side */
  size?: Size | Rectangular | Variable;

  /** The content of the margin wrapper */
  children?: React.ReactNode;

  /** Prop to override any style if necessary, use sparingly */
  style?: Style;

  /** Reponsive prop overrides */
  responsive?: Responsive<this>;
}

export const Padding = ({ responsive, ...rest }: PaddingProps) => {
  const { children, size: rawSize, style } = useResponsiveProps<PaddingProps>(rest, responsive);

  const isUniform = typeof rawSize !== 'object';

  const size = !isUniform && _isRectangular(rawSize) ? _computeSize(rawSize) : rawSize;

  return (
    <div
      className={cx(styles.root, {
        [styles[camelCase(size as Size) as keyof typeof styles]]: isUniform && size != null,
        [styles.resetPadding]: !isUniform,
        [styles[camelCase(`${(size as Variable)?.left}_LEFT`) as keyof typeof styles]]:
          !isUniform && (size as Variable)?.left != null,
        [styles[camelCase(`${(size as Variable)?.right}_RIGHT`) as keyof typeof styles]]:
          !isUniform && (size as Variable)?.right != null,
        [styles[camelCase(`${(size as Variable)?.top}_TOP`) as keyof typeof styles]]:
          !isUniform && (size as Variable)?.top != null,
        [styles[camelCase(`${(size as Variable)?.bottom}_BOTTOM`) as keyof typeof styles]]:
          !isUniform && (size as Variable)?.bottom != null,
      })}
      style={style}>
      {children}
    </div>
  );
};
