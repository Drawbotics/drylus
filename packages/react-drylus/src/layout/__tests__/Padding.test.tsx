import React from 'react';
import { create } from 'react-test-renderer';

import { Size } from '../../enums';
import { Padding } from '../Padding';

describe('Padding', () => {
  describe('matches snapshot when', () => {
    it('has a uniform size', () => {
      const tree = create(<Padding size={Size.LARGE}>Content</Padding>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a size only on 1 side', () => {
      const tree = create(<Padding size={{ left: Size.LARGE }}>Content</Padding>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a different size on each side', () => {
      const tree = create(
        <Padding
          size={{
            left: Size.LARGE,
            top: Size.SMALL,
            bottom: Size.DEFAULT,
            right: Size.EXTRA_SMALL,
          }}>
          Content
        </Padding>,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has vertical and horizontal sizes', () => {
      const tree = create(
        <Padding
          size={{
            vertical: Size.LARGE,
            horizontal: Size.SMALL,
          }}>
          Content
        </Padding>,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
