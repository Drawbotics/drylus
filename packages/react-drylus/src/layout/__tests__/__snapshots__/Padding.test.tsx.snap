// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Padding matches snapshot when has a different size on each side 1`] = `
.emotion-0 {
  padding: 24px;
  padding: 0;
  padding-left: 32px;
  padding-right: 8px;
  padding-top: 16px;
  padding-bottom: 24px;
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding: 16px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding: 0;
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding-left: 24px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding-right: calc(8px / 2);
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding-top: 8px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding-bottom: 16px;
  }
}

<div
  className="emotion-0"
>
  Content
</div>
`;

exports[`Padding matches snapshot when has a size only on 1 side 1`] = `
.emotion-0 {
  padding: 24px;
  padding: 0;
  padding-left: 32px;
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding: 16px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding: 0;
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding-left: 24px;
  }
}

<div
  className="emotion-0"
>
  Content
</div>
`;

exports[`Padding matches snapshot when has a uniform size 1`] = `
.emotion-0 {
  padding: 24px;
  padding: 32px;
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding: 16px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding: 24px;
  }
}

<div
  className="emotion-0"
>
  Content
</div>
`;

exports[`Padding matches snapshot when has vertical and horizontal sizes 1`] = `
.emotion-0 {
  padding: 24px;
  padding: 0;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 32px;
  padding-bottom: 32px;
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding: 16px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding: 0;
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding-left: 8px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding-right: 8px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding-top: 24px;
  }
}

@media only screen and (max-width:768px) {
  .emotion-0 {
    padding-bottom: 24px;
  }
}

<div
  className="emotion-0"
>
  Content
</div>
`;
